File: go/problems/day9_test.go
package problems

import "testing"

var day9 Solvable[int64] = Day9{}

func TestDay9_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day9_1")
	result := day9.Solve(input)

	var expected int64 = 114
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay9_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day9_2")
	result := day9.Solve(input)

	var expected int64 = 2043183816
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day9.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day9.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 9", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day9_1");
    const result = solve(input);
    assert.equal(result, 114);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day9_2");
    const result = solve(input);
    assert.equal(result, 2043183816);
  });
});
